repository/src/main/java/ru/karalionak/rest/query/OrderedViewQuery.java

```diff
@@ -0,0 +1,80 @@
+package ru.karalionak.rest.query;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OrderedViewQuery {
+    private static final QueryHelper queryHelper = new QueryHelper();
+    private final StringBuilder query;
+    private final List<Object> values;
+
+    {
+        values = new ArrayList<>();
+        query = new StringBuilder("SELECT * FROM gift_certificates " +
+                "JOIN gift_certificate_tag gct on gift_certificates.id = gct.certificate_id " +
+                "JOIN tags on tags.id = gct.tag_id");
+    }
+
+    public OrderedViewQuery(StringBuilder query, List<Object> values) {
+        this.query.append(query);
+        this.values.addAll(values);
+    }
+
+    public String getQuery() {
+        return query.toString();
+    }
+
+    public List<Object> getValues() {
+        return values;
+    }
+
+    public static QueryBuilder builder(){
+        return new QueryBuilder();
+    }
+
+    public static class QueryBuilder{
+        private final StringBuilder query;
+        private final List<Object> values;
+        private boolean conditionFlag;
```

Comment:
Let's simplify your logic. Don't use condition flag. Store 'tagName' and 'regex' in builder and append query only in build() method.
Also, it will protect you from case when service calls sortingQuery(SortingContext sortingContext) before filteringMethods.